### DISCUSSION TITLE
Does streaming work with grpc-web?

### DISCUSSION THREAD

**Discussion by louis030195** [hyperium/tonic#792] [Q&A] [answered]
Hi and thanks for the wonderful lib!

I experimented this with it https://github.com/louis030195/codex-code-translation,

Might be a noob question but, I am unsure if client-side streaming is supported with grpc-web?

For example, in https://github.com/improbable-eng/grpc-web#client-side-streaming it does not seem, maybe I'm missing some foundational knowledge.

**Comment by davidpdrsn** [comment, marked as answer]
From [the docs](https://docs.rs/tonic-web/0.1.0/tonic_web/#limitations)

> Currently, grpc-web clients can only perform unary and server-streaming calls. These are the only requests this crate is designed to handle. Support for client and bi-directional streaming will be officially supported when clients do.
